app/account/components/AccountDetails/components/AvatarUpdate/AvatarUpdate.tsx

```diff
@@ -0,0 +1,173 @@
+"use client"
+
+import { updateUserInfo } from "@api"
+import { Button, ModalOverlay } from "@components"
+import { User } from "@interfaces"
+import { useUserStore } from "@stores"
+import { customFetch } from "@utils"
+import React, { useCallback, useState } from "react"
+import Cropper from "react-easy-crop"
+import "./avatarUpdate.scss"
+import { getCroppedImg } from "./cropUtils"
+
+const AvatarUpdate = () => {
+  const fetchUserData = useUserStore((state) => state.fetchUserData)
+  const [imageSrc, setImageSrc] = useState<string | null>(null)
+  const [crop, setCrop] = useState({ x: 0, y: 0 })
+  const [zoom, setZoom] = useState(1)
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState<null | { width: number; height: number; x: number; y: number }>(null)
+  const [isCropping, setIsCropping] = useState(false)
+  const [userMsg, setUserMsg] = useState<string>()
+  const fileInputRef = React.useRef<HTMLInputElement | null>(null)
+  const user = useUserStore((state) => state.user)
+  const { username } = (user as User) ?? {}
+  const msgShowDuration = 2000
+
+  const closeModal = () => {
+    // reset cropper
+    setImageSrc(null)
+    setCrop({ x: 0, y: 0 })
+    setZoom(1)
+    setCroppedAreaPixels(null)
+    setUserMsg(undefined)
+
+    if (fileInputRef.current) {
+      fileInputRef.current.value = ""
+    }
+  }
+
+  const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+
+    if (file) {
+      const reader = new FileReader()
+
+      reader.onload = () => setImageSrc(reader.result as string)
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const handleCropComplete = useCallback((croppedArea: any, croppedAreaPixels: any) => {
+    setCroppedAreaPixels(croppedAreaPixels)
+  }, [])
```

Comment:
The empty dependency array will also trigger another es-lint warning (see #59). Is the use of `useCallback` necessary here?

Reply:
No use whatsover. I have to check codepilots code better in the future

Reply:
Happens to the best of us! No worries.